atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/TargetedSweeper.java

```diff
@@ -229,44 +217,62 @@ private void assertInitialized() {
         }
     }
 
+    private int getPartitionBatchLimit() {
+        return runtime.get().enableAutoTuning()
+                ? Integer.MAX_VALUE
+                : runtime.get().maximumPartitionsToBatchInSingleRead();
+    }
+
     private class BackgroundSweepScheduler implements AutoCloseable {
         private final int numThreads;
         private final SweeperStrategy sweepStrategy;
         private final AtomicLong counter = new AtomicLong(0);
+        private final SweepDelay delay;
 
         private ScheduledExecutorService executorService;
 
         private BackgroundSweepScheduler(int numThreads, SweeperStrategy sweepStrategy) {
             this.numThreads = numThreads;
             this.sweepStrategy = sweepStrategy;
+            this.delay = new SweepDelay(runtime.get().pauseMillis());
         }
 
         private void scheduleBackgroundThreads() {
             if (numThreads > 0 && executorService == null) {
                 executorService = PTExecutors
                         .newScheduledThreadPoolExecutor(numThreads, new NamedThreadFactory("Targeted Sweep", true));
```

Comment:
Something to be aware of (I guess it's fine for now but will affect the next stage of changes): this will spin up `numThreads` many core threads and core thread timeout is not allowed by the executor returned here, so some changes will be needed when we extend this to cover that.

Reply:
Yeah, once we do task scaling, this is set to 1 instead